### PR TITLE
[Resolving Bug #291]: Corrected if block.

### PR DESCRIPTION
**Pull request by ChinoCodeDemon**
# Related Issues
Resolving the coin update 0 issue #291 

# Summary of Changes
changed boolean check in line 26 of src/commandDetails/coin/update.ts to a typecheck.

# Steps to Reproduce
coin update 0 should now work correctly.

I want to additionally say, that i just guessed that the attributes are converted before passing them into commands.
Tell me if i did a mistake.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,7 @@ yarn-error.log
 **/.DS_Store
 /config/dev
 /tmp
+
+# editor
+.devcontainer
+.vscode
```
Comment (Picowchew) [inline .gitignore:15]: We would like to keep the VS Code workspace settings, hence `.vscode` has not been added to `.gitignore`.

File: src/commandDetails/coin/update.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ const coinUpdateExecuteCommand: SapphireMessageExecuteType = async (client, mess
 
   // Second mandatory argument is amount
   const amount = args['amount'];
-  if (!amount) {
+  if (typeof amount != 'number') {
```
Comment (Picowchew) [inline src/commandDetails/coin/update.ts:26]: `!==` should be used instead of `!=`.

**Review by Picowchew — CHANGES_REQUESTED** [review]
Thanks for the PR! I left a few small comments. Also, it would be nice if you could include a screenshot in the PR description to show that updating to 0 coins works.

File: src/commandDetails/coin/update.ts (hunk 2 above)
Comment (ChinoCodeDemon) [reply src/commandDetails/coin/update.ts:26]: well at that line a change would only affect readability.

**Comment by ChinoCodeDemon** [comment]
are coin commands disabled by default?

**Review by Picowchew — APPROVED** [review]
LGTM!

Are you receiving an error for the coin commands? I checked out your branch and the coin commands seemed to be working fine for me.

**Comment by ChinoCodeDemon** [comment]
Well any command I try doesn't output anything, the only command that gives
me output is `blackjack`.

**Comment by Picowchew** [comment]
Do any errors show when you do `yarn logs`?

**Comment by ChinoCodeDemon** [comment]
Well I ran it without docker container but in a visual studio code
devcontainer. (Which is similar to a code space container)
And the terminal didn't show anything on commands, only the usual stuff.
Is worth looking into.
I usually use devcontainer for projects I just find via issue search.

**Comment by Picowchew** [comment]
Do you mind showing your terminal output? In addition, do you want to try using Docker instead?

**Comment by ChinoCodeDemon** [comment]
There ae some exceptions surrounding `./config/dev/vars.json`
And i don't see the `coin` command registered in there.
```
10:07:55 AM - Starting compilation in watch mode...

10:07:58 AM - Found 0 errors. Watching for file changes.
Error when loading '/workspaces/codeybot/dist/src/commands/interviewer/interviewer.js': Error: ENOENT: no such file or directory, open './config/dev/vars.json'
    at Object.openSync (node:fs:599:3)
    at readFileSync (node:fs:467:35)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/config.js:6:49)
    at Module._compile (node:internal/modules/cjs/loader:1112:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/utils/embeds.js:6:18) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './config/dev/vars.json'
}
Error when loading '/workspaces/codeybot/dist/src/commands/coffeeChat/coffeeChat.js': Error: ENOENT: no such file or directory, open './config/dev/vars.json'
    at Object.openSync (node:fs:599:3)
    at readFileSync (node:fs:467:35)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/config.js:6:49)
    at Module._compile (node:internal/modules/cjs/loader:1112:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/components/coffeeChat.js:10:18) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './config/dev/vars.json'
}
Error when loading '/workspaces/codeybot/dist/src/commands/coin/coin.js': Error: ENOENT: no such file or directory, open './config/dev/vars.json'
    at Object.openSync (node:fs:599:3)
    at readFileSync (node:fs:467:35)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/config.js:6:49)
    at Module._compile (node:internal/modules/cjs/loader:1112:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/utils/embeds.js:6:18) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './config/dev/vars.json'
}
Error when loading '/workspaces/codeybot/dist/src/commands/profile/profile.js': Error: ENOENT: no such file or directory, open './config/dev/vars.json'
    at Object.openSync (node:fs:599:3)
    at readFileSync (node:fs:467:35)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/config.js:6:49)
    at Module._compile (node:internal/modules/cjs/loader:1112:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/utils/embeds.js:6:18) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './config/dev/vars.json'
}
Error when loading '/workspaces/codeybot/dist/src/listeners/messageCreate.js': Error: ENOENT: no such file or directory, open './config/dev/vars.json'
    at Object.openSync (node:fs:599:3)
    at readFileSync (node:fs:467:35)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/config.js:6:49)
    at Module._compile (node:internal/modules/cjs/loader:1112:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/listeners/messageCreate.js:16:18) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './config/dev/vars.json'
}
Error when loading '/workspaces/codeybot/dist/src/listeners/ready.js': Error: ENOENT: no such file or directory, open './config/dev/vars.json'
    at Object.openSync (node:fs:599:3)
    at readFileSync (node:fs:467:35)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/config.js:6:49)
    at Module._compile (node:internal/modules/cjs/loader:1112:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/components/coffeeChat.js:10:18) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './config/dev/vars.json'
}
Error when loading '/workspaces/codeybot/dist/src/commands/suggestion/suggestion.js': Error: ENOENT: no such file or directory, open './config/dev/vars.json'
    at Object.openSync (node:fs:599:3)
    at readFileSync (node:fs:467:35)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/config.js:6:49)
    at Module._compile (node:internal/modules/cjs/loader:1112:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/workspaces/codeybot/dist/src/utils/embeds.js:6:18) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './config/dev/vars.json'
}
2022-08-18 10:08:00 - INFO  - ApplicationCommandRegistries: Initializing...
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[info] Preparing to process 1 possible command registrations / updates...
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[info] Registering id "1008928699992768593" to internal chat input map
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[info] Checking if command "info" is identical with global chat input command with id "1008928699992768593"
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[info] Took 0ms to process differences
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[info] Command "info" is identical to command "info" (1008928699992768593)
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[flipcoin] Preparing to process 1 possible command registrations / updates...
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[flipcoin] Registering id "1008928701364314122" to internal chat input map
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[flipcoin] Checking if command "flipcoin" is identical with global chat input command with id "1008928701364314122"
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[flipcoin] Took 0ms to process differences
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[flipcoin] Command "flipcoin" is identical to command "flipcoin" (1008928701364314122)
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[rolldice] Preparing to process 1 possible command registrations / updates...
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[rolldice] Registering id "1008928702698098728" to internal chat input map
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[rolldice] Checking if command "rolldice" is identical with global chat input command with id "1008928702698098728"
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[rolldice] Took 1ms to process differences
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[rolldice] Command "rolldice" is identical to command "rolldice" (1008928702698098728)
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[ping] Preparing to process 1 possible command registrations / updates...
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[ping] Registering id "1008370318018084925" to internal chat input map
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[ping] Checking if command "ping" is identical with global chat input command with id "1008370318018084925"
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[ping] Took 0ms to process differences
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[ping] Command "ping" is identical to command "ping" (1008370318018084925)
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[help] Preparing to process 1 possible command registrations / updates...
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[help] Registering id "1008928704778473502" to internal chat input map
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[help] Checking if command "help" is identical with global chat input command with id "1008928704778473502"
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[help] Took 0ms to process differences
2022-08-18 10:08:00 - DEBUG - ApplicationCommandRegistry[help] Command "help" is identical to command "help" (1008928704778473502)
2022-08-18 10:08:00 - INFO  - ApplicationCommandRegistries: Took 409ms to initialize.
(node:614) DeprecationWarning: Passing strings for MessageEmbed#setFooter is deprecated. Pass a sole object instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

**Comment by mcpenguin** [comment]
It seems that your `config/dev/vars.json` was not copied over to your dev container. In the Dockerfile we automatically do this - you might want to automatically set up something similar for your dev container.

**Comment by ChinoCodeDemon** [comment]
> It seems that your `config/dev/vars.json` was not copied over to your dev container. In the Dockerfile we automatically do this - you might want to automatically set up something similar for your dev container.

I will when i try to solve another issue on this project.
My most honest answer would be that I pretty much search for easy to solve issues that do not affect the time I use for my own projects.

Since most bugs are caused by badly written checks, the time I usually spend on those is about 3 minutes and most of the time spent is just trying to get the exact position of the incorrect check.
In rare cases such problems could arise if the input is parsed incorrectly or a calculation might be badly written.

If you guys find another bug that might be caused by such stuff you could mention me, that makes it easier for me to get something new to fix.
I won't fix any database issues, since such stuff needs excessive testing in most cases.

**Pull request merged**